build-system/tasks/check-owners.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const deglob = require('globs-to-files');
+const log = require('fancy-log');
+const path = require('path');
+const {getOutput} = require('../common/exec');
+const {green, cyan, red, yellow} = require('ansi-colors');
+const {highlight} = require('cli-highlight');
+const {isTravisBuild} = require('../common/travis');
+
+const rootDir = path.dirname(path.dirname(__dirname));
+const prettierCmd =
+  'node_modules/.bin/prettier --parser json5 --config .prettierrc';
+
+/**
+ * Prettier exit codes. See https://prettier.io/docs/en/cli.html#exit-codes
+ * @enum {number}
+ */
+const PrettierResult = {
+  SUCCESS: 0,
+  FAILURE: 1,
+  ERROR: 2,
+};
+
+/**
+ * Checks and optionally fixes the formatting in OWNERS files using Prettier.
+ * The cumulative result is returned to the `gulp` process via process.exitCode
+ * so that all OWNERS files can be checked / fixed.
+ */
+async function checkOwners() {
```

Comment:
This feels odd to me because this task doesn't check the validity of OWNERS file, just the formatting. Wouldn't formatting checks go into `gulp lint`?

The actual owners check needs to reference the amp-github-apps repo, IIRC.

Reply:
The `gulp lint` task uses `eslint` to check `.js` files. It turns out that `eslint` is incapable of parsing non-JS files like JSON5 formatted `OWNERS` files. See https://github.com/ampproject/amphtml/pull/24993#issue-326386826 for more details.

Re: using the parser from the `amp-github-apps` repo, you raise a good point. Here's what I'm envisioning for `gulp check-owners`:
1. It should be responsible for checking all `OWNERS` file changes to the `amphtml` repo
2. It should run as part of the `amphtml` Travis checks
3. It should check for formatting / style (and block PRs if there are errors)
4. It should check for correctness (and block PRs if there are errors)

IIRC, this summarizes what we discussed a few weeks ago. Assuming you agree with the above list, this PR implements items 1-3, with item 4 being gated by the availability of a parsing API from `amp-github-apps`. In addition, I don't believe the parsing API will check for or enforce items 1-3, so I think this is the right place to implement them.

WDYT? Would it help to rename the functions in this file and add a TODO for the missing parsing check to make things clearer?

Reply:
I agree with 1, 2, and 4, especially since they were points I was pushing for.

My question was just whether this check belongs in a separate formatting check. But since we don't see a json format check in the future, I'm ok with keeping the format check in `gulp owners-check` with the rename/TODO change you suggested. Thanks!

Reply:
Update: Renamed `checkFile` to `checkFormatting` and added a TODO for `checkCorrectness`.

Reply:
Further update after discussing this with @estherkim: The check for json / json5 formatting should be separated from the check for `OWNERS` correctness. This way, we can make the former a more general purpose check that can apply to other files in the repo, and allow for auto-fixing via prettier. The latter can be implemented using the owners bot parsing API.

Coming up in the next few commits.